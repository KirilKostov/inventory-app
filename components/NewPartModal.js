import React, { useState } from "react";
import { Modal, View, StyleSheet } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { Button } from "./Button";
import { Input } from "./Input";
import { TitleText } from "./TitleText";
import Colors from "../constants/colors";

export const NewPartModal = (props) => {
  const [name, setName] = useState(null);
  const [barcode, setBarcode] = useState(null);
  return (
    <Modal
      visible={props.modalVisible}
      animationType="slide"
      statusBarTranslucent
    >
      <View style={styles.container}>
        <View style={styles.content}>
          <TitleText style={styles.title}>
            <MaterialCommunityIcons name="cog" size={24} color="black" /> Add
            new part
          </TitleText>
          <Input
            placeholder="name"
            style={styles.input}
            onChangeText={(name) => setName(name)}
            autoCorrect={false}
          />
          <Input
            placeholder="barcode"
            style={styles.input}
            onChangeText={(barcode) => setBarcode(barcode)}
          />
          <View style={styles.buttonContainer}>
            <Button
              title="Save"
              style={{ backgroundColor: Colors.green, width: "45%" }}
              onPress={() => props.handleAddRecord({ name, barcode })}
              disabled={!name || !barcode}
            />
            <Button
              title="Cancel"
              style={{ backgroundColor: Colors.lightGray, width: "45%" }}
              textStyle={{ color: "black" }}
              onPress={props.onClose}
            />
          </View>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  content: {
    width: "80%",
    alignItems: "center",
  },
  title: {
    marginVertical: 40,
    alignItems: "center",
    justifyContent: "center",
  },
  input: {
    width: "100%",
  },
  buttonContainer: {
    marginVertical: 40,
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-between",
  },
});
