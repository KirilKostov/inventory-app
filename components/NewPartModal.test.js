import React from "react";
import renderer from "react-test-renderer";
import { NewPartModal } from "./NewPartModal";

describe("<NewPartModal />", () => {
  it("renders correctly", () => {
    const testRenderer = renderer
      .create(<NewPartModal modalVisible />)
      .toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});
