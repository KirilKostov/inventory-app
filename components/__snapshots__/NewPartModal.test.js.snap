// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPartModal /> renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  statusBarTranslucent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontFamily": "gibson-semibold",
            "fontSize": 22,
            "justifyContent": "center",
            "marginVertical": 40,
          }
        }
      >
         Add new part
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="name"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#78757c",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 40,
            "marginVertical": 10,
            "textAlign": "center",
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="barcode"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#78757c",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 40,
            "marginVertical": 10,
            "textAlign": "center",
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 40,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "#0067ff",
                "borderRadius": 4,
                "elevation": 3,
                "justifyContent": "center",
                "paddingHorizontal": 32,
                "paddingVertical": 12,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.4,
                "shadowRadius": 2,
              },
              Object {
                "backgroundColor": "#6bbd57",
                "width": "45%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "gibson-regular",
                "fontSize": 16,
                "letterSpacing": 0.25,
                "lineHeight": 21,
              }
            }
          >
            Save
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "#0067ff",
                "borderRadius": 4,
                "elevation": 3,
                "justifyContent": "center",
                "paddingHorizontal": 32,
                "paddingVertical": 12,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.4,
                "shadowRadius": 2,
              },
              Object {
                "backgroundColor": "#e9e9e9",
                "width": "45%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "gibson-regular",
                "fontSize": 16,
                "letterSpacing": 0.25,
                "lineHeight": 21,
              }
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
